src/shared/meshes/cell_linked_list.hpp

```diff
@@ -44,4 +44,108 @@ void CellLinkedList::searchNeighborsByParticles(
                  });
 }
 //=================================================================================================//
+template <class LocalDynamicsFunction>
+void CellLinkedList::particle_for_split(const execution::SequencedPolicy &, const LocalDynamicsFunction &local_dynamics_function)
+{
+    // foward sweeping
+    for (size_t k = 0; k < number_of_split_cell_lists_; k++)
+    {
+        // get the corresponding 2D/3D split cell index (m, n)
+        // e.g., for k = 0, split_cell_index = (0,0), for k = 3, split_cell_index = (1,0), etc.
+        const Arrayi split_cell_index = transfer1DtoMeshIndex(3 * Arrayi::Ones(), k);
```

Comment:
I have transformed the method to a 1D loop, but recomputing the corresponding cell indices every time can increase the computational cost a bit.
@DongWuTUM

Reply:
@WeiyiVirtonomy how about the overall speed?

Reply:
The speed looks close to the old method. For test_3d_arch, the computational time on my machine is:
Old method: without damping 26.1719s, with damping 37.9359s, damping time 11.764s
New method: without damping 29.2173s, with damping 42.0216s, damping time 12.8043s

Reply:
It looks good.